internal/services/controller/handlers/handler.go

```diff
@@ -0,0 +1,69 @@
+package handlers
+
+import (
+	"reflect"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	"castai-agent/internal/castai"
+	"castai-agent/internal/services/controller/delta"
+	"castai-agent/internal/services/controller/handlers/transformers"
+	"castai-agent/internal/services/controller/handlers/transformers/cleaner"
+	"castai-agent/internal/services/controller/handlers/transformers/deletedfinalstateunknown"
+)
+
+type handler struct {
+	log          logrus.FieldLogger
+	handledType  reflect.Type
+	queue        workqueue.Interface
+	filters      Filters
+	transformers transformers.Transformers
+}
+type Handler interface {
+	cache.ResourceEventHandler
+}
+
+func NewHandler(log logrus.FieldLogger, queue workqueue.Interface, handledType reflect.Type) Handler {
+	return &handler{
+		log:         log,
+		handledType: handledType,
+		queue:       queue,
+		filters:     Filters{},
+		transformers: transformers.Transformers{
+			deletedfinalstateunknown.New(),
+			cleaner.New(),
+		},
+	}
+}
+
+func (h *handler) handle(e castai.EventType, obj interface{}) {
+	if !h.filters.apply(e, obj) {
+		return
+	}
+
+	e, obj = h.transformers.Apply(e, obj)
+
```

Comment:
is this + the required infrastucture of setting up the lists of filters/transformers and their interfaces more understandable than directly writting here like this:

```suggestion
can we use the opportunity to change the pattern here? this "plugins" code style is excessive when you can just do a

```go
   detelefinalstateunknown.Delete(apply)
   cleaner.Clean(obj)
```

Reply:
I may be opinionated, but this was the whole idea of this refactoring, having an easy way to add data transformations/filtering. This will allow us to add custom types (provided they are added to the schema) to the delta.